incubator/hnc/e2e/tests/issues_test.go

```diff
@@ -36,4 +43,45 @@ var _ = Describe("Issues", func() {
 		runShouldContain("No conditions", 1, "kubectl hns describe", nsParent)
 		runShouldContain("No conditions", 1, "kubectl hns describe", nsChild)
 	})
+
+	It("Should test cascadingDelete in HC and anchor reconcilers - issue #501", func() {
+		// Setting up a 3-level tree with 'parent' as the root
+		mustRun("kubectl create ns", nsParent)
+		mustRun("kubectl create ns", nsSub1)
+		mustRun("kubectl hns set", nsSub1, "--parent", nsParent)
+		mustRun("kubectl annotate ns", nsSub1, "hnc.x-k8s.io/subnamespaceOf=" + nsParent)
```

Comment:
Add a comment here to say that this subnamespace is created as a full namespace with `subnamespaceOf` annotation and parent set later just to create a subnamespace without anchor.

Reply:
done